pandas/tseries/tdi.py

```diff
@@ -754,7 +758,7 @@ def _maybe_cast_slice_bound(self, label, side, kind):
     def _get_string_slice(self, key, use_lhs=True, use_rhs=True):
         freq = getattr(self, 'freqstr',
                        getattr(self, 'inferred_freq', None))
-        if is_integer(key) or is_float(key):
+        if is_integer(key) or is_float(key) or key is tslib.NaT:
```

Comment:
further we don't use lib.is_integer ever directly 
import from pandas.types.common (as it is)